packages/cactus-plugin-ledger-connector-iroha/src/main/json/openapi.json

```diff
@@ -0,0 +1,429 @@
+          "To get available permissions per role in the system.",
+          "To get a specific block, using its height as an identifier.",
+          "To retrieve a receipt of a CallEngine command. Allows to access the event log created during computations inside the EVM.",
+          "To get new blocks as soon as they are committed, a user can invoke FetchCommits RPC call to Iroha network.",
+          "A query that returns a list of peers in Iroha network."
+        ],
+        "x-enum-varnames": [
+          "GetAccount",
+          "GetAccountDetail",
+          "GetAssetInfo",
+          "GetAccountAssets",
+          "GetTransactions",
+          "GetPendingTransactions",
+          "GetAccountTransactions",
+          "GetAccountAssetTransactions",
+          "GetRoles",
+          "GetSignatories",
+          "GetRolePermissions",
+          "GetBlock",
+          "GetEngineReceipts",
+          "FetchCommits",
+          "GetPeers"
+        ]
+      },
+      "KeyPair": {
+        "type": "object",
+        "required": ["publicKey", "privateKey"],
+        "properties": {
+          "publicKey": {
+            "type": "string",
+            "nullable": false
+          },
+          "privateKey": {
+            "type": "string",
+            "nullable": false
+          }
+        }
+      },
+      "RunTransactionRequest": {
+        "type": "object",
+        "required": ["commandName", "params"],
+        "properties": {
+          "commandName": {
+            "type": "string",
+            "nullable": false
+          },
+          "baseConfig": {
+            "$ref": "#/components/schemas/IrohaBaseConfig",
+            "nullable": false
+          },
+          "params": {
+            "description": "The list of arguments to pass in to the transaction request.",
+            "type": "array",
+            "default": [],
+            "items": {}
+          }
+        }
+      },
+      "IrohaBaseConfig": {
+        "type": "object",
+        "additionalProperties": true,
+        "properties": {
+          "irohaHost": {
+            "type": "string",
+            "nullable": false
+          },
+          "irohaPort": {
+            "type": "number",
+            "nullable": false
+          },
+          "creatorAccountId": {
+            "type": "string",
+            "nullable": false
+          },
+          "privKey": {
+            "type": "array",
+            "items": {},
+            "default": [],
+            "nullable": false
+          },
+          "quorum": {
+            "type": "number",
+            "nullable": false
+          },
+          "timeoutLimit": {
+            "type": "number",
+            "nullable": false
+          }
+        }
+      },
+      "RunTransactionResponse": {
+        "type": "object",
+        "required": ["transactionReceipt"],
+        "properties": {
+          "transactionReceipt": {}
+        }
+      },
+      "InvokeContractV1Request": {
```

Comment:
@hxlaf Because the contract invocation is not supported I'd refactor this for now (and anything else related to it that appears as dead code until we actually have contract invocation supported)

(the other option is to gut it at least, so that it doesn't have ethereum specific parameteres danling in it probably the most appropriate is to keep it on and empty out the parameters in it because we don't yet know what those will look like - this will enable the endpoint to remain implemented but with the http not implemented error being a fixed response)

Reply:
I am unsure whether you wish it to be refactored in this form: (i.e., leaving very minimal properties)
![image](https://user-images.githubusercontent.com/10954157/129981410-57917732-e979-4485-bfc3-a8656fe0d217.png)

Reply:
@hxlaf If You want (and have time) You can take a look into current PR (not merged yet) of another intern who is extending smart contracts: https://github.com/hyperledger/iroha/pull/1326/files - as You see they are not called directly by name, but the name is converted somehow according to documentation (function `get_first_four_bytes_of_keccak()`)